packages/web/src/scss/components/Grid/_tools.scss

```diff
@@ -7,14 +7,37 @@
 // * $gutters: the gutters map
 // * $breakpoints: the breakpoints map
 @mixin gutters($gutters, $breakpoints) {
+    // stylelint-disable custom-property-no-missing-var-function -- We are only building the fallback value here.
+    $column-property-name: --grid-spacing-x;
+    $row-property-name: --grid-spacing-y;
+    // stylelint-enable
+
+    $column-fallback: map.get($gutters, 'mobile');
+    $row-fallback: map.get($gutters, 'mobile');
+
     @each $breakpoint-name, $breakpoint-value in $breakpoints {
         @if not map.has-key($gutters, $breakpoint-name) {
             @error 'Invalid breakpoint specified! #{$breakpoint-name} doesn\'t exist. Use one of #{map.keys($breakpoints)}';
         }
+
+        @if $breakpoint-value > 0 {
+            // stylelint-disable custom-property-no-missing-var-function -- We are only building the fallback value here.
+            $column-property-name: --grid-spacing-x-#{$breakpoint-name};
+            $row-property-name: --grid-spacing-y-#{$breakpoint-name};
+            // stylelint-enable
+        }
+
         @include breakpoint.up($breakpoint-value) {
-            gap: map.get($gutters, $breakpoint-name);
+            --grid-column-gap: var(#{$column-property-name}, #{$column-fallback});
+            --grid-row-gap: var(#{$row-property-name}, #{$row-fallback});
         }
+
+        $column-fallback: var(#{$column-property-name}, #{$column-fallback});
+        $row-fallback: var(#{$row-property-name}, #{$row-fallback});
     }
+
+    grid-column-gap: var(--grid-column-gap);
+    grid-row-gap: var(--grid-row-gap);
```

Comment:
We also need something like this
```
.Stack--hasSpacing > * {
    @each $breakpoint-name, $breakpoint-value in theme.$breakpoints {
        @if $breakpoint-value > 0 {
            --stack-spacing-#{$breakpoint-name}: initial;
        } @else {
            --stack-spacing: initial;
        }
    }
}
```

Otherwise, nested grid gaps will be influenced by its parent.